ui/app/routes/jobs/run/index.js

```diff
@@ -1,34 +1,59 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 import classic from 'ember-classic-decorator';
+import notifyForbidden from 'nomad-ui/utils/notify-forbidden';
 
 @classic
 export default class RunRoute extends Route {
   @service can;
+  @service router;
   @service store;
   @service system;
 
+  queryParams = {
+    template: {
+      refreshModel: true,
+    },
+  };
+
   beforeModel(transition) {
     if (
       this.can.cannot('run job', null, {
         namespace: transition.to.queryParams.namespace,
       })
     ) {
-      this.transitionTo('jobs');
+      this.router.transitionTo('jobs');
     }
   }
 
-  model() {
-    // When jobs are created with a namespace attribute, it is verified against
-    // available namespaces to prevent redirecting to a non-existent namespace.
-    return this.store.findAll('namespace').then(() => {
+  async model({ template }) {
+    try {
+      // When jobs are created with a namespace attribute, it is verified against
+      // available namespaces to prevent redirecting to a non-existent namespace.
+      await this.store.findAll('namespace');
+
+      // If template is set in URL, create the job model and add the definition
+      if (template) {
+        const templateRecord = await this.store.findRecord(
+          'variable',
+          template
+        );
+
+        return this.store.createRecord('job', {
+          _newDefinition: templateRecord.items.template,
+        });
+      }
```

Comment:
And a question: should we nullify the queryParam for template once it's loaded? Once loaded, its job is pretty much done, right? Better to have a clean URL?

Reply:
At this layer, the `error` object that is returned is an assertion message (`string`) and a stack trace.

If we want to trigger a toast notification upon receiving a `404` error we'll want to update the `handleResponse` method on the `Variable Adapter` which would have ramifications for the `Variable` views who aren't handling `404`'s like this. It might just cause a slight inconsistency with error message handling but I'll handle it there and we can do some manual testing and see what happens.

```
message
: 
"Assertion Failed: You made a 'findRecord' request for a 'variable' with id 'Auto-conflicting Varible@default', but the adapter's response did not have any data"
stack
: 
"Error: Assertion Failed: You made a 'findRecord' request for a 'variable' with id 'Auto-conflicting Varible@default', but the adapter's re....
```

Reply:
> Once loaded, its job is pretty much done, right? Better to have a clean URL?

IMO, we care about URL's because the contract the user has with the browser. If the user hits refresh on the page, they'll get a different output if we clear the query parameter. And it'll impact the back/forward buttons on the browser.

I'd probably keep the query parameter.

Reply:
![image](https://user-images.githubusercontent.com/41024828/211386347-7a63df42-a30c-4687-b071-9c68ae0e0b3f.png)